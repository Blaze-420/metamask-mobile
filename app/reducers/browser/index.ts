import { type BrowserAction, BrowserActionTypes } from '../../actions/browser';
import { type BrowserState } from './types';
import AppConstants from '../../core/AppConstants';

export * from './types';

/**
 * Initial browser state
 */
export const initialBrowserState: BrowserState = {
  history: [],
  whitelist: [],
  tabs: [],
  favicons: [],
  activeTab: null,
  // Keep track of viewed Dapps, which is used for MetaMetricsEvents.DAPP_VIEWED event
  visitedDappsByHostname: {},
};

/**
 * Browser reducer
 */
/* eslint-disable @typescript-eslint/default-param-last */
const browserReducer = (
  state: BrowserState = initialBrowserState,
  action: BrowserAction,
): BrowserState => {
  switch (action.type) {
    case BrowserActionTypes.ADD_TO_VIEWED_DAPP: {
      const { hostname } = action;
      return {
        ...state,
        visitedDappsByHostname: {
          ...state.visitedDappsByHostname,
          [hostname]: true,
        },
      };
    }
    case BrowserActionTypes.ADD_TO_BROWSER_HISTORY: {
      const { url, name } = action;

      return {
        ...state,
        history: [...state.history, { url, name }].slice(0, 50),
      };
    }
    case BrowserActionTypes.ADD_TO_BROWSER_WHITELIST: {
      const { url } = action;

      return {
        ...state,
        whitelist: [...state.whitelist, url],
      };
    }
    case BrowserActionTypes.CLEAR_BROWSER_HISTORY: {
      return {
        ...state,
        history: [],
        favicons: [],
        tabs: [{ url: AppConstants.HOMEPAGE_URL, id: action.id }],
        activeTab: action.id,
      };
    }
    case BrowserActionTypes.CLOSE_ALL_TABS: {
      return {
        ...state,
        tabs: [],
      };
    }
    case BrowserActionTypes.CREATE_NEW_TAB: {
      const { url, linkType, id } = action;

      return {
        ...state,
        tabs: [
          ...state.tabs,
          {
            url,
            id,
            ...(linkType && { linkType }),
          },
        ],
      };
    }
    case BrowserActionTypes.CLOSE_TAB: {
      const { id } = action;

      return {
        ...state,
        tabs: state.tabs.filter((tab) => tab.id !== id),
      };
    }
    case BrowserActionTypes.SET_ACTIVE_TAB: {
      const { id } = action;

      return {
        ...state,
        activeTab: id,
      };
    }
    case BrowserActionTypes.UPDATE_TAB: {
      return {
        ...state,
        tabs: state.tabs.map((tab) => {
          if (tab.id === action.id) {
            return { ...tab, ...action.data };
          }
          return { ...tab };
        }),
      };
    }
    case BrowserActionTypes.STORE_FAVICON_URL: {
      const { origin, url } = action;

      return {
        ...state,
        favicons: [
          { origin: action.origin, url: action.url },
          ...state.favicons,
        ].slice(0, AppConstants.FAVICON_CACHE_MAX_SIZE),
      };
    }
    default:
      return state;
  }
};
export default browserReducer;
