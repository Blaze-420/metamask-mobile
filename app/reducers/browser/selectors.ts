import { RootState } from '..';

export const selectBrowserState = (state: RootState) => state.browser;

export const selectBrowserHistory = (state: RootState) => state.browser.history;

/**
 * Gets the selected search engine from the Redux state
 * @param state - Redux state
 * @returns - Selected search engine
 */
export const selectSearchEngine = (state: RootState) =>
  state.settings.searchEngine;
