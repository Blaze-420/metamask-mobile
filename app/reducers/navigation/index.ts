import { NavigationActionType } from '../../actions/navigation/types';
import { NavigationState } from './types';

export * from './types';

export * from './selectors';

/**
 * Initial navigation state
 */
const initialState: NavigationState = {
  currentRoute: 'WalletView',
  currentBottomNavRoute: 'Wallet',
};

/**
 * Navigation reducer
 */
// TODO: Replace "any" with type
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const navigationReducer = (state = initialState, action: any = {}) => {
  switch (action.type) {
    case NavigationActionType.SET_CURRENT_ROUTE:
      return {
        ...state,
        currentRoute: action.payload.route,
      };
    case NavigationActionType.SET_CURRENT_BOTTOM_NAV_ROUTE:
      return {
        ...state,
        currentBottomNavRoute: action.payload.route,
      };
    default:
      return state;
  }
};

/**
 * Selectors
 */
// TODO: Replace "any" with type
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const getCurrentRoute = (state: any) => state.navigation.currentRoute;
// TODO: Replace "any" with type
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const getCurrentBottomNavRoute = (state: any) =>
  state.navigation.currentBottomNavRoute;

export default navigationReducer;
