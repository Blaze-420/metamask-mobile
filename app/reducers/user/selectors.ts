import { RootState } from '..';

/**
 * Selects the user state
 */
export const selectUserState = (state: RootState) => state.user;

/**
 * Selects the userLoggedIn state
 */
export const selectUserLoggedIn = (state: RootState) => state.user.userLoggedIn;
