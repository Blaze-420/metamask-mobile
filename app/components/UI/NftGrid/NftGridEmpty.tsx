import React from 'react';
import { View, TouchableOpacity, Image } from 'react-native';
import { strings } from '../../../../locales/i18n';
import Text, {
  TextColor,
  TextVariant, // TextColor,
  // TextVariant,
} from '../../../component-library/components/Texts/Text';
import styleSheet from './NftGrid.styles';
import { useStyles } from '../../hooks/useStyles';
import { StackNavigationProp } from '@react-navigation/stack';
import AppConstants from '../../../core/AppConstants';
import { WalletViewSelectorsIDs } from '../../../../e2e/selectors/wallet/WalletView.selectors';

import noNftPlaceholderSrc from '../../../images/no-nfts-placeholder.png';
interface NftGridNavigationParamList {
  AddAsset: { assetType: string };
  [key: string]: undefined | object;
}

interface NftGridProps {
  navigation: StackNavigationProp<NftGridNavigationParamList, 'AddAsset'>;
}

function NftGridEmpty({ navigation }: NftGridProps) {
  const { styles } = useStyles(styleSheet, {});

  return (
    <View
      // eslint-disable-next-line react-native/no-inline-styles
      style={{
        display: 'flex',
        alignItems: 'center',
      }}
    >
      <Image
        // eslint-disable-next-line react-native/no-color-literals, react-native/no-inline-styles
        style={{
          height: 90,
          width: 90,
          tintColor: 'lightgray',
          marginTop: 30,
          marginBottom: 12,
        }}
        source={noNftPlaceholderSrc}
        resizeMode="contain"
      />
      <Text
        style={styles.headingMd}
        variant={TextVariant.HeadingMD}
        color={TextColor.Alternative}
      >
        {strings('wallet.no_nfts_yet')}
      </Text>
      <TouchableOpacity
        onPress={() =>
          navigation.navigate('Webview', {
            screen: 'SimpleWebview',
            params: { url: AppConstants.URLS.NFT },
          })
        }
        testID={WalletViewSelectorsIDs.IMPORT_NFT_BUTTON}
      >
        <Text
          variant={TextVariant.BodyMDMedium}
          color={TextColor.Info}
          onPress={() => console.log('goToLearnMore')}
        >
          {strings('wallet.learn_more')}
        </Text>
      </TouchableOpacity>
    </View>
  );
}

export default NftGridEmpty;
