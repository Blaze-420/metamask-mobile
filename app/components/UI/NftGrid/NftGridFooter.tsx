import React, { useCallback } from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { strings } from '../../../../locales/i18n';
import Text, {
  TextColor,
  TextVariant,
} from '../../../component-library/components/Texts/Text';
import { StackNavigationProp } from '@react-navigation/stack';
import { WalletViewSelectorsIDs } from '../../../../e2e/selectors/wallet/WalletView.selectors';

interface NftGridNavigationParamList {
  AddAsset: { assetType: string };
  [key: string]: undefined | object;
}

interface NftGridFooterProps {
  navigation: StackNavigationProp<NftGridNavigationParamList, 'AddAsset'>;
}

const styles = StyleSheet.create({
  footerContainer: {
    display: 'flex',
    alignItems: 'center',
  },
});

function NftGridFooter({ navigation }: NftGridFooterProps) {
  const handlePress = useCallback(
    () => navigation.push('AddAsset', { assetType: 'collectible' }),
    [navigation],
  );

  return (
    <View style={styles.footerContainer}>
      <Text variant={TextVariant.BodyMDMedium} color={TextColor.Alternative}>
        {strings('wallet.no_collectibles')}
      </Text>
      <TouchableOpacity
        onPress={handlePress}
        disabled={false}
        testID={WalletViewSelectorsIDs.IMPORT_NFT_BUTTON}
      >
        <Text variant={TextVariant.BodyMDMedium} color={TextColor.Info}>
          {strings('wallet.add_collectibles')}
        </Text>
      </TouchableOpacity>
    </View>
  );
}

export default NftGridFooter;
