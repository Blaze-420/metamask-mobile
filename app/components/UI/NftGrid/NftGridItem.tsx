import React, { useCallback, useRef, useState } from 'react';
import {
  View,
  TouchableOpacity,
  Alert,
  ScrollView,
  ActivityIndicator,
  Image,
  FlatList,
  RefreshControl,
} from 'react-native';
import { useSelector } from 'react-redux';
import CollectibleMedia from '../CollectibleMedia';
import ActionSheet from '@metamask/react-native-actionsheet';
import { strings } from '../../../../locales/i18n';
import Engine from '../../../core/Engine';
import { removeFavoriteCollectible } from '../../../actions/collectibles';
import {
  collectiblesSelector,
  isNftFetchingProgressSelector,
} from '../../../reducers/collectibles';
import { useTheme } from '../../../util/theme';
import Text, {
  TextColor,
  TextVariant,
} from '../../../component-library/components/Texts/Text';
import {
  MetaMetricsEvents,
  useMetrics,
} from '../../../components/hooks/useMetrics';
import { getDecimalChainId } from '../../../util/networks';
import { Nft } from '@metamask/assets-controllers';
import { debounce } from 'lodash';
import styleSheet from './NftGrid.styles';
import { useStyles } from '../../hooks/useStyles';
import { StackNavigationProp } from '@react-navigation/stack';
import { WalletViewSelectorsIDs } from '../../../../e2e/selectors/wallet/WalletView.selectors';
import CollectibleDetectionModal from '../CollectibleDetectionModal';
import { selectUseNftDetection } from '../../../selectors/preferencesController';
import ButtonLink from '../../../component-library/components/Buttons/Button/variants/ButtonLink';
import AppConstants from '../../../core/AppConstants';
import {
  RefreshTestId,
  SpinnerTestId,
} from '../CollectibleContracts/constants';

const noNftPlaceholderSrc = require('../../../images/no-nfts-placeholder.png');

const debouncedNavigation = debounce((navigation, collectible) => {
  navigation.navigate('NftDetails', { collectible });
}, 200);

interface ActionSheetType {
  show: () => void;
}

interface LongPressedCollectibleType {
  address: string;
  tokenId: string;
}

interface NftGridNavigationParamList {
  AddAsset: { assetType: string };
  [key: string]: undefined | object;
}

function NftGridItem({
  nft,
  navigation,
}: {
  nft: Nft;
  navigation: StackNavigationProp<NftGridNavigationParamList, 'AddAsset'>;
}) {
  const actionSheetRef = useRef<ActionSheetType>(null);
  const longPressedCollectible = useRef<LongPressedCollectibleType | null>(
    null,
  );
  const { styles } = useStyles(styleSheet, {});
  const { trackEvent, createEventBuilder } = useMetrics();

  const onRefresh = useCallback(async () => {
    requestAnimationFrame(async () => {
      setRefreshing(true);
      const { NftDetectionController, NftController } = Engine.context;
      const actions = [
        NftDetectionController.detectNfts(),
        NftController.checkAndUpdateAllNftsOwnershipStatus(),
      ];
      await Promise.allSettled(actions);
      setRefreshing(false);
    });
  }, [setRefreshing]);

  const onLongPressCollectible = useCallback((collectible) => {
    actionSheetRef?.current?.show();
    longPressedCollectible.current = collectible;
  }, []);

  const removeNft = () => {
    const { NftController } = Engine.context;

    if (
      !longPressedCollectible?.current?.address &&
      !longPressedCollectible?.current?.tokenId
    ) {
      return null;
    }

    removeFavoriteCollectible(
      selectedAddress,
      chainId,
      longPressedCollectible.current,
    );
    NftController.removeAndIgnoreNft(
      longPressedCollectible.current.address,
      longPressedCollectible.current.tokenId,
    );

    trackEvent(
      createEventBuilder(MetaMetricsEvents.COLLECTIBLE_REMOVED)
        .addProperties({
          chain_id: getDecimalChainId(chainId),
        })
        .build(),
    );
    Alert.alert(
      strings('wallet.collectible_removed_title'),
      strings('wallet.collectible_removed_desc'),
    );
  };

  const refreshMetadata = () => {
    const { NftController } = Engine.context;

    if (
      !longPressedCollectible?.current?.address &&
      !longPressedCollectible?.current?.tokenId
    ) {
      return null;
    }

    NftController.addNft(
      longPressedCollectible.current.address,
      longPressedCollectible.current.tokenId,
    );
  };

  const handleMenuAction = (index: number) => {
    if (index === 1) {
      removeNft();
    } else if (index === 0) {
      refreshMetadata();
    }
  };

  const onItemPress = useCallback(
    (nftItem) => {
      debouncedNavigation(navigation, nftItem);
    },
    [navigation],
  );

  if (!nft) return null;
  return (
    <TouchableOpacity
      key={nft.address}
      style={styles.collectibleCard}
      onPress={() => onItemPress(nft)}
      onLongPress={() => onLongPressCollectible(nft)}
      testID={nft.name as string}
    >
      <CollectibleMedia
        style={styles.collectibleIcon}
        collectible={nft}
        isTokenImage
      />
      <Text numberOfLines={1} ellipsizeMode="tail">
        {nft.name}
      </Text>
      <Text numberOfLines={1} ellipsizeMode="tail">
        {nft.collection?.name}
      </Text>
    </TouchableOpacity>
  );
}

export default NftGrid;
