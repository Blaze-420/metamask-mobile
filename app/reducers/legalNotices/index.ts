import { RootState } from '..';
import { Action } from 'redux';
import ACTIONS from './types';
import { selectRemoteFeatureFlags } from '../../selectors/featureFlagController';

const initialState = {
  newPrivacyPolicyToastClickedOrClosed: false,
  newPrivacyPolicyToastShownDate: null,
};

export const storePrivacyPolicyShownDate = (timestamp: number) => ({
  type: ACTIONS.STORE_PRIVACY_POLICY_SHOWN_DATE,
  payload: timestamp,
});

export const storePrivacyPolicyClickedOrClosed = () => ({
  type: ACTIONS.STORE_PRIVACY_POLICY_CLICKED_OR_CLOSED,
});

// New selector for privacy policy feature flag
export const selectPrivacyPolicyUpdateFeatureFlag = (state: RootState): string => {
  const remoteFeatureFlags = selectRemoteFeatureFlags(state);
  const policyValue = String(remoteFeatureFlags?.transactionsPrivacyPolicyUpdate || '');

  if (policyValue === 'active_update') {
    return '2025-04-01T12:00:00Z'; // A date in the past
  }

  return policyValue;
};

export const shouldShowNewPrivacyToastSelector = (state: RootState): boolean => {
  const {
    newPrivacyPolicyToastShownDate,
    newPrivacyPolicyToastClickedOrClosed,
  } = state.legalNotices;

  // Get the feature flag value
  const policyUpdateDate = selectPrivacyPolicyUpdateFeatureFlag(state);

  // If feature flag is not set or empty, don't show toast
  if (!policyUpdateDate) return false;

  // If toast was already clicked or closed, don't show it again
  if (newPrivacyPolicyToastClickedOrClosed) return false;

  const currentDate = new Date();
  const policyDate = new Date(policyUpdateDate);

  // Check if current date is after policy update date
  const isPastPolicyDate = currentDate >= policyDate;

  // Check if toast was shown recently (within 24 hours)
  const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
  const isRecent = newPrivacyPolicyToastShownDate
    ? currentDate.getTime() - newPrivacyPolicyToastShownDate < oneDayInMilliseconds
    : false;

  return (
    isPastPolicyDate &&
    (!newPrivacyPolicyToastShownDate || isRecent)
  );
};

export interface LegalNoticesAction extends Action {
  newPrivacyPolicyToastShownDate: boolean;
  payload: number;
}

export const isPastPrivacyPolicyDate = (state: RootState): boolean => {
  const policyUpdateDate = selectPrivacyPolicyUpdateFeatureFlag(state);
  if (!policyUpdateDate) return false;

  const currentDate = new Date();
  const policyDate = new Date(policyUpdateDate);

  return currentDate >= policyDate;
};

const legalNoticesReducer = (
  state = initialState,
  action: LegalNoticesAction = {
    type: '',
    newPrivacyPolicyToastShownDate: false,
    payload: 0,
  },
) => {
  switch (action.type) {
    case ACTIONS.STORE_PRIVACY_POLICY_SHOWN_DATE: {
      if (state.newPrivacyPolicyToastShownDate !== null) {
        return state;
      }

      return {
        ...state,
        newPrivacyPolicyToastShownDate: action.payload,
      };
    }

    case ACTIONS.STORE_PRIVACY_POLICY_CLICKED_OR_CLOSED: {
      return { ...state, newPrivacyPolicyToastClickedOrClosed: true };
    }

    default:
      return state;
  }
};

export default legalNoticesReducer;
