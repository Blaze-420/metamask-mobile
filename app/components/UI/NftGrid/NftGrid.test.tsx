import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import NftGridFooter from './NftGridFooter';
import { WalletViewSelectorsIDs } from '../../../../e2e/selectors/wallet/WalletView.selectors';
import { StackNavigationProp } from '@react-navigation/stack';
import NftGridEmpty from './NftGridEmpty';
import NftGridItem from './NftGridItem';
import { Nft } from '@metamask/assets-controllers';
import renderWithProvider, {
  DeepPartial,
} from '../../../util/test/renderWithProvider';
import { RootState } from '../../../reducers';
import { useNavigation } from '@react-navigation/native';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

type MockNavigation = StackNavigationProp<
  {
    AddAsset: { assetType: string };
    [key: string]: object | undefined;
  },
  'AddAsset'
>;

const mockNavigation: MockNavigation = {
  push: jest.fn(),
  navigate: jest.fn(),
  goBack: jest.fn(),
  pop: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  setParams: jest.fn(),
  getParent: jest.fn(),
} as unknown as MockNavigation;

const mockState: DeepPartial<RootState> = {
  engine: {
    backgroundState: {
      PreferencesController: {
        isIpfsGatewayEnabled: true,
      },
    },
  },
} as unknown as RootState;

describe('NftGridFooter', () => {
  it('renders without crashing', () => {
    const { getByText } = render(<NftGridFooter navigation={mockNavigation} />);
    expect(getByText('Donâ€™t see your NFT?')).toBeTruthy();
    expect(getByText('Import NFTs')).toBeTruthy();
  });

  it('calls navigation.push when the button is pressed', () => {
    const { getByTestId } = render(
      <NftGridFooter navigation={mockNavigation} />,
    );
    const button = getByTestId(WalletViewSelectorsIDs.IMPORT_NFT_BUTTON);
    fireEvent.press(button);
    expect(mockNavigation.push).toHaveBeenCalledWith('AddAsset', {
      assetType: 'collectible',
    });
  });

  it('matches the snapshot', () => {
    const tree = render(<NftGridFooter navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('NftGridEmpty', () => {
  it('renders without crashing', () => {
    const { getByText, getByTestId } = render(
      <NftGridEmpty navigation={mockNavigation} />,
    );
    expect(getByText('No NFTs yet')).toBeTruthy();
    expect(getByText('Learn more')).toBeTruthy();
    expect(getByTestId(WalletViewSelectorsIDs.IMPORT_NFT_BUTTON)).toBeTruthy();
  });

  it('calls navigation.navigate when the button is pressed', () => {
    const { getByTestId } = render(
      <NftGridEmpty navigation={mockNavigation} />,
    );
    const button = getByTestId(WalletViewSelectorsIDs.IMPORT_NFT_BUTTON);
    fireEvent.press(button);
    expect(mockNavigation.navigate).toHaveBeenCalledWith('Webview', {
      screen: 'SimpleWebview',
      params: {
        url: 'https://support.metamask.io/nfts/nft-tokens-in-your-metamask-wallet/',
      },
    });
  });

  it('logs "goToLearnMore" when the learn_more text is pressed', () => {
    const consoleSpy = jest.spyOn(console, 'log');
    const { getByText } = render(<NftGridEmpty navigation={mockNavigation} />);
    const learnMoreText = getByText('Learn more');
    fireEvent.press(learnMoreText);

    // TODO: actually test for learn more redirect
    expect(consoleSpy).toHaveBeenCalledWith('goToLearnMore');
    consoleSpy.mockRestore();
  });

  it('matches the snapshot', () => {
    const tree = render(<NftGridEmpty navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('NftGridItem', () => {
  const mockNft: Nft = {
    address: '0x123',
    tokenId: '1',
    name: 'Test NFT',
    image: 'https://example.com/image.png',
    collection: {
      name: 'Test Collection',
    },
    description: '',
    standard: 'erc721',
  };

  const mockNavigate = jest.fn();
  (useNavigation as jest.Mock).mockReturnValue({
    navigate: mockNavigate,
    goBack: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    reset: jest.fn(),
    popToTop: jest.fn(),
  });

  it('renders correctly with a valid nft', () => {
    const { getByText, getByTestId } = renderWithProvider(
      <NftGridItem nft={mockNft} navigation={mockNavigation} />,
      { state: mockState },
    );

    expect(getByTestId(mockNft.name as string)).toBeTruthy();
    expect(getByText('Test NFT')).toBeTruthy();
    expect(getByText('Test Collection')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderWithProvider(
      <NftGridItem nft={mockNft} navigation={mockNavigation} />,
      { state: mockState },
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
