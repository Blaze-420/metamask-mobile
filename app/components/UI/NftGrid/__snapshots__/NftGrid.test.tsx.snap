// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NftGridEmpty matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      {
        "height": 90,
        "marginBottom": 12,
        "marginTop": 30,
        "tintColor": "lightgray",
        "width": 90,
      }
    }
  />
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#6a737d",
        "fontFamily": "EuclidCircularB-Bold",
        "fontSize": 18,
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 24,
        "marginTop": 10,
      }
    }
  >
    No NFTs yet
  </Text>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text
      accessibilityRole="text"
      onPress={[Function]}
      style={
        {
          "color": "#0376c9",
          "fontFamily": "EuclidCircularB-Medium",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      Learn more
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`NftGridFooter matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#6a737d",
        "fontFamily": "EuclidCircularB-Medium",
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0,
        "lineHeight": 22,
      }
    }
  >
    Donâ€™t see your NFT?
  </Text>
  <TouchableOpacity
    disabled={false}
    onPress={[Function]}
    testID="import-collectible-button"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#0376c9",
          "fontFamily": "EuclidCircularB-Medium",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      Import NFTs
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`NftGridItem matches the snapshot 1`] = `
<TouchableOpacity
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    {
      "alignItems": "center",
      "flexBasis": "33%",
      "justifyContent": "center",
      "marginBottom": 10,
      "padding": 10,
    }
  }
  testID="Test NFT"
>
  <View
    style={
      {
        "backgroundColor": undefined,
        "borderRadius": 12,
        "flex": 0,
      }
    }
  >
    <View
      useNativeDriver={true}
    >
      <View
        onLoadEnd={[Function]}
      >
        <View
          onLayout={[Function]}
          style={
            {
              "alignSelf": "flex-start",
              "position": "relative",
            }
          }
          testID="badge-wrapper-badge"
        >
          <View>
            <View
              style={
                [
                  {
                    "borderRadius": 12,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {
                    "aspectRatio": 1,
                    "width": "100%",
                  },
                ]
              }
            >
              <Image
                fadeIn={true}
                isTokenImage={true}
                onError={[Function]}
                resizeMode="cover"
                source={
                  {
                    "uri": "https://example.com/image.png",
                  }
                }
                style={
                  {
                    "borderRadius": 8,
                    "height": "100%",
                    "width": "100%",
                  }
                }
                testID="nft-image"
              />
            </View>
          </View>
          <View
            style={
              {
                "position": "absolute",
                "right": -4,
                "top": -4,
              }
            }
          >
            <View
              onLayout={[Function]}
              style={{}}
              testID="badgenetwork"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#ffffff",
                    "borderColor": "#ffffff",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "shadowColor": "#0000001a",
                    "shadowOffset": {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 4,
                    "width": 16,
                  }
                }
              >
                <Image
                  onError={[Function]}
                  resizeMode="contain"
                  source={1}
                  style={
                    {
                      "height": 16,
                      "width": 16,
                    }
                  }
                  testID="network-avatar-image"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            [
              {},
              {
                "backgroundColor": "#eee",
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <Text
    accessibilityRole="text"
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "#141618",
        "fontFamily": "EuclidCircularB-Regular",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 22,
      }
    }
  >
    Test NFT
  </Text>
  <Text
    accessibilityRole="text"
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "#141618",
        "fontFamily": "EuclidCircularB-Regular",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 22,
      }
    }
  >
    Test Collection
  </Text>
</TouchableOpacity>
`;
